lib/feedsme/index.js

```diff
@@ -54,153 +64,223 @@ function pkgName(pkg) {
  * @param {Fucntion} done Completion callback.
  * @api private
  */
-Feedsme.prototype.change = function change(env, data, done) {
-  var pkg = this.extractLatest(data);
-  var app = this.app;
-  //
-  // TODO: Figure out if we can get previous here or if we need to do a GET on
-  // artifactory which would kind of suck
-  //
-
-  var meta = {
-    name: pkg.name,
-    version: pkg.version,
-    env: env
-  };
+Feedsme.prototype.change = async function change(env, data) {
+  const { Dependent, DependentOf } = this.models;
+  const pkg = this.extractLatest(data);
+  const publish = env === 'dev';
+
+  const { name, version } = pkg;
+
+  const meta = { name, version, env };
+
+  this.log.info(`Processing change for ${ pkgName(pkg) } in ${ env }`, meta);
+  // Update dependent and dependent_of models for reference
+  await this.resolve(env, pkg);
+
+  // Fetch things we need to figure out what exactly to do
+  const [releaseLine, dep, depOf] = await Promise.all([
+    publish ? this.release.head({ pkg: name }) : this.release.get({ pkg: name, version }),
+    Dependent.get(pkg.name),
+    DependentOf.get(pkg.name)
+  ]);
+
+  await Promise.all([
+    // Root or Main package triggers dependent builds
+    this.triggerDep({ env, releaseLine, pkg, dep }),
+    // Dependent packages associate with their release-line, only on publish
+    publish && this.triggerDepOf({ pkg, depOf })
+  ].filter(Boolean));
+
+  this.log.info(`Successfully processed change for ${ pkgName(pkg) } in ${ env }`, meta);
+};
 
-  function finish(err) {
-    if (err) return done(err);
+Feedsme.prototype.triggerDepOf = async function triggerDepOf({ pkg, depOf }) {
+  if (!depOf || (depOf && !depOf.dependentOf)) return this.log.info(`${pkg.name} does not have a parent package it depends on`);
 
-    app.log.info(`Successfully processed change for ${ pkgName(pkg) } in ${ env }`, meta);
-    done.apply(done, arguments);
-  }
+  // Fetch the release line for the parent package that owns it
+  const line = await this.release.get({ pkg: depOf.dependentOf });
```

Comment:
I wonder if this should (maybe at some point?) need to get the latest semver-matching release line of the parent package. e.g. package A relies on B@^1.0.0 and shouldn't attach itself to the B@2 release lines.